cneuromax/projects/friends_language_encoder/peftmodule.py

```diff
@@ -21,21 +20,25 @@ class PEFTLitModule(BaseLitModule, ABC):
 
     def __init__(
         self: "PEFTLitModule",
-        config: BaseLitModuleConfig,
-        nnmodule: nn.Module,
-        optimizer: partial[Optimizer],
-        scheduler: partial[LRScheduler],
+        config: "BaseLitModuleConfig",
+        peft_config: "PeftConfig",
+        *args: Any,  # noqa: ANN401, ARG002
+        **kwargs: Any,  # noqa: ANN401, ARG002
     ) -> None:
         """."""
-        super().__init__(config, nnmodule, optimizer, scheduler)
+        super().__init__(
+            config,
+            peft_config,
+            *args,
+            **kwargs,
+        )
         self.config = config
-        peft_config = LoraConfig(
-            task_type=self.config.task_type,
-            lora_alpha=self.config.lora_alpha,
-            r=self.config.local_rank,
-            inference_mode=self.config.inference_mode,
-            lora_dropout=self.config.lora_dropout,
+        self.peft_config = peft_config
+
+        lora_config = LoraConfig(
+            task_type=self.peft_config.task_type,
+            lora_alpha=self.peft_config.lora_alpha,
+            r=self.peft_config.lora_rank,
+            lora_dropout=self.peft_config.lora_dropout,
         )
-        self.nnmodule = get_peft_model(nnmodule, peft_config)
-        self.optimizer_partial = optimizer
-        self.scheduler_partial = scheduler
+        self.nnmodule = get_peft_model(self.nnmodule, lora_config)
```

Comment:
you don't need to reinstantiate a LoraConfig object. LoraConfig is a PeftConfig. If this doesn't make complete sense, brush up on python inheritance

Reply:
all of this can be replaced by `self.nnmodule = get_peft_model(self.nnmodule, peft_config)`